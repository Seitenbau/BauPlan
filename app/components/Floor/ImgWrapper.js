import styled from 'styled-components';

const ImgWrapper = styled.div`
  padding: ${(props) => props.theme.plans.image.padding};
  position: relative;
`;

export default ImgWrapper;
