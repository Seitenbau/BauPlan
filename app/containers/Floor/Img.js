import styled from 'styled-components';

import NormalImg from 'components/Img';

const Img = styled(NormalImg)`
  width: ${(props) => props.theme.plans.image.width}vw;
  height: auto;
  margin: 0 auto;
`;

export default Img;
