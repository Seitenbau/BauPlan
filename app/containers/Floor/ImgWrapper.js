import styled from 'styled-components';

const ImgWrapper = styled.div`
  position: relative;
`;
// padding: ${(props) => props.theme.plans.image.padding};

export default ImgWrapper;
