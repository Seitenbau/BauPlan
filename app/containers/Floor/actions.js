import {
  FLOOR_RENDERED,
} from './constants';

export function floorRendered(data) {
  return {
    type: FLOOR_RENDERED,
    data,
  };
}
