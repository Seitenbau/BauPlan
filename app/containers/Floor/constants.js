export const FLOOR_RENDERED = 'app/Floor/FLOOR_RENDERED';
