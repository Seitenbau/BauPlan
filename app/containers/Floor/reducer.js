
/*
 *
 * Plans reducer
 *
 */

import { fromJS } from 'immutable';
import {
  FLOOR_RENDERED,
} from './constants';

const initialState = fromJS({
  name: '',
});

function floorReducer(state = initialState, action) {
  switch (action.type) {
    case FLOOR_RENDERED:
      return state.set('name');
    default:
      return state;
  }
}

export default floorReducer;
