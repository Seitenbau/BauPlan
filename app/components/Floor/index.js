/**
*
* Floor
*
*/

import React from 'react';
import { PropTypes } from 'prop-types';
import imageLoader from 'utils/ImageLoader';
import RoomLabel from 'components/RoomLabel';
import Table from 'containers/Table';
import { flatten } from 'utils/helper';
import Img from './Img';
import StyledFloor from './StyledFloor';
import ImgWrapper from './ImgWrapper';
import Label from './label';

class Floor extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  find(id) {
    const ids = flatten([id]);
    if (typeof id !== 'undefined') {
      return this.props.projects.filter((e) => ids.indexOf(e.id) > -1);
    }
    return [];
  }
  render() {
    const floorPlanImage = imageLoader(this.props.imageName);
    return (
      <StyledFloor innerRef={(ref) => { this.ref = ref; }}>
        <Label>{this.props.name}</Label>
        <ImgWrapper >
          <Img id={this.props.id} name={this.props.name} src={floorPlanImage} alt={`Floorplan ${this.props.name}`} />
          {this.props.labels.map((label, i) =>
            <RoomLabel
              key={i}
              name={label.name}
              left={label.x}
              top={label.y}
            />)}
          {this.props.tables.map((table, j) =>
            <Table
              key={j}
              className={`table table-${j}`}
              name={table.name}
              number={table.number}
              projects={this.find(table.projects)}
              x={table.x}
              y={table.y}
              rotation={table.rotation}
            />)}

        </ImgWrapper>
      </StyledFloor>
    );
  }
}

Floor.propTypes = {
  name: PropTypes.string.isRequired,
  imageName: PropTypes.string.isRequired,
  labels: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]),
  id: PropTypes.string,
  tables: PropTypes.array,
  projects: PropTypes.array,
};

export default Floor;
