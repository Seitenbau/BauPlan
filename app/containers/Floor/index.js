/**
*
* Floor
*
*/

import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
// import { createStructuredSelector } from 'reselect';
import imageLoader from 'floorplans';
import RoomLabel from 'components/RoomLabel';
import Table from 'containers/Table';
import { flatten } from 'utils/helper';
import makeSelectScaleFactor from './selectors';
import Img from './Img';
import StyledFloor from './StyledFloor';
import ImgWrapper from './ImgWrapper';
import Label from './label';

class Floor extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  find(id) {
    const ids = flatten([id]);
    if (typeof id !== 'undefined') {
      return this.props.projects.filter((e) => ids.indexOf(e.id) > -1);
    }
    return [];
  }
  render() {
    const floorPlanImage = imageLoader(this.props.imageName);
    return (
      <StyledFloor innerRef={(ref) => { this.ref = ref; }}>
        <Label>{this.props.name}</Label>
        <ImgWrapper >
          <Img mapScaleFactor={this.props.mapScaleFactor} name={this.props.name} scale={this.props['width-height']} src={floorPlanImage} alt={`Floorplan ${this.props.name}`} />
          {this.props.labels.map((label, i) =>
            <RoomLabel
              key={i}
              name={label.name}
              left={label.x}
              top={label.y}
            />)}

          {this.props.tables.map((table, j) =>
            <Table
              key={j}
              scaleFactor={this.props.scaleFactor}
              className={`table table-${j}`}
              name={table.name}
              number={table.number}
              projects={this.find(table.projects)}
              x={table.x}
              y={table.y}
              rotation={table.rotation}
            />)}

        </ImgWrapper>
      </StyledFloor>
    );
  }
}

Floor.propTypes = {
  name: PropTypes.string.isRequired,
  imageName: PropTypes.string.isRequired,
  labels: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]),
  tables: PropTypes.array,
  projects: PropTypes.array,
  'width-height': PropTypes.string,
  mapScaleFactor: PropTypes.number,
  scaleFactor: PropTypes.number,
};

Floor.defaultProps = {
  scaleFactor: 1,
};

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const makeMapStateToProps = () => {
  const selectScaleFactor = makeSelectScaleFactor();
  /* eslint-disable */
  const mapStateToProps = (state, props) => {
    return {
      scaleFactor: selectScaleFactor(state, props),
    }
  };
  /* eslint-enable */
  return mapStateToProps;
};


export default connect(makeMapStateToProps, mapDispatchToProps)(Floor);
