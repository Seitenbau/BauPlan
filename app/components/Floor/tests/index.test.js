// import React from 'react';
import { shallow } from 'enzyme';

import Floor from '../index';

describe('<Floor />', () => {
  it('Expect to Throw', () => {
    expect(() => shallow(<Floor></Floor>)).toThrow();
  });
});
