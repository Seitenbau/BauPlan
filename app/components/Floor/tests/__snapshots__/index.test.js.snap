exports[`<Floor /> Expect to render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.h2 />
    <styled.div>
        <Styled(undefined)
            src={undefined} />
        <RoomLabel
            left={0}
            name="Room 1"
            top={0} />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.h2 />
      <styled.div>
            <Styled(undefined)
                  src={undefined} />
            <RoomLabel
                  left={0}
                  name="Room 1"
                  top={0} />
      </styled.div>
</styled.div>,
  ],
  "options": Object {
    "context": Object {
      "__styled-components__": [Function],
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {
        "__styled-components__": [Function],
      },
      "_currentElement": <Floor
        labels={
                Array [
                        Object {
                          "name": "Room 1",
                          "x": 0,
                          "y": 0,
                        },
                      ]
        } />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Floor {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "labels": Array [
            Object {
              "name": "Room 1",
              "x": 0,
              "y": 0,
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.h2 />
          <styled.div>
                    <Styled(undefined)
                              src={undefined} />
                    <RoomLabel
                              left={0}
                              name="Room 1"
                              top={0} />
          </styled.div>
</styled.div>,
        "_debugID": 4,
        "_renderedOutput": <styled.div>
          <styled.h2 />
          <styled.div>
                    <Styled(undefined)
                              src={undefined} />
                    <RoomLabel
                              left={0}
                              name="Room 1"
                              top={0} />
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Floor
    labels={
        Array [
            Object {
              "name": "Room 1",
              "x": 0,
              "y": 0,
            },
          ]
    } />,
}
`;
